File: SchoolData/src/main/java/org/gb/homework/schooldata/model/factories/impls/TeacherFactory.java
package org.gb.homework.schooldata.model.factories.impls;

import org.gb.homework.schooldata.model.Teacher;
import org.gb.homework.schooldata.model.factories.UserFactory;

public class TeacherFactory implements UserFactory<Teacher> {

    @Override
    public Teacher deserialize(String line) {
        var strings = line.split(";\t;");

        int id = Integer.parseInt(strings[1]);
        boolean isActual = Boolean.parseBoolean(strings[2]);
        String name = strings[3];

        Teacher teacher = new Teacher(id, name);
        teacher.setActual(isActual);

        return teacher;
    }
}

File: csharp/SchoolData/Model/Factories/Impls/TeacherFactory.cs
namespace Gb.Homework.SchoolData.Model.Factories.Impls
{

    public class TeacherFactory : UserFactory<Teacher>
    {

        public Teacher Deserialize(string line)
        {
            var strings = line.Split(";\t;");

            int id = int.Parse(strings[1]);
            bool isActual = bool.Parse(strings[2]);
            var name = strings[3];

            var teacher = new Teacher(id, name)
            {
                IsActual = isActual
            };

            return teacher;
        }
    }
}
